### PR TITLE
[code-utils] use `num_utils` `Min`/`Max` to simplify code

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR uses `Min` and `Max` from num_utils to replace Ternary operators.

### CODE REVIEW

File: src/core/backbone_router/bbr_local.cpp [hunk 1]
```diff
@@ -252,8 +252,7 @@ void Local::HandleBackboneRouterPrimaryUpdate(Leader::State aState, const Backbo
 
         if (!Get<Mle::MleRouter>().IsLeader())
         {
-            delay += Random::NonCrypto::GetUint8InRange(0, mRegistrationJitter < 255 ? mRegistrationJitter + 1
-                                                                                     : mRegistrationJitter);
+            delay += Random::NonCrypto::GetUint8InRange(0, Min(mRegistrationJitter + 1, 255));
```
Comment (zhanglongxia) [inline src/core/backbone_router/bbr_local.cpp:255]: If `mRegistrationJitter ` equals to 300, the result of the original code is 300. The result of the new code is 255.

**Review by zhanglongxia — APPROVED** [review]
LGTM

File: src/core/crypto/crypto_platform.cpp [hunk 2]
```diff
@@ -733,7 +733,7 @@ OT_TOOL_WEAK void otPlatCryptoPbkdf2GenerateKey(const uint8_t *aPassword,
             }
         }
 
-        useLen = (keyLen < kBlockSize) ? keyLen : kBlockSize;
+        useLen = static_cast<uint16_t>(Min(static_cast<size_t>(keyLen), kBlockSize));
```
Comment (abtink) [inline src/core/crypto/crypto_platform.cpp:736]: Same?

File: src/core/thread/mlr_manager.cpp [hunk 3]
```diff
@@ -648,8 +648,7 @@ void MlrManager::UpdateReregistrationDelay(bool aRereg)
         {
             // Calculate renewing period according to Thread Spec. 5.24.2.3.2
             // The random time t SHOULD be chosen such that (0.5* MLR-Timeout) < t < (MLR-Timeout – 9 seconds).
-            effectiveMlrTimeout = config.mMlrTimeout > Mle::kMlrTimeoutMin ? config.mMlrTimeout
-                                                                           : static_cast<uint32_t>(Mle::kMlrTimeoutMin);
+            effectiveMlrTimeout = Max(config.mMlrTimeout, static_cast<uint32_t>(Mle::kMlrTimeoutMin));
```
Comment (abtink) [inline src/core/thread/mlr_manager.cpp:651]: `kMlrTimeoutMin` seems to be `uint32_t`, remove cast?

**Review by abtink — COMMENTED** [review]
Thanks @Irving-cl. This is a good simplification. 👍 
Some smaller suggestions below:

File: src/lib/spinel/radio_spinel_impl.hpp [hunk 4]
```diff
@@ -2588,7 +2588,7 @@ uint32_t RadioSpinel<InterfaceType, ProcessContextType>::Snprintf(char *aDest, u
     len = vsnprintf(aDest, static_cast<size_t>(aSize), aFormat, args);
     va_end(args);
 
-    return (len < 0) ? 0 : (static_cast<uint32_t>(len) > aSize - 1 ? aSize - 1 : static_cast<uint32_t>(len));
+    return (len < 0) ? 0 : Min(static_cast<uint32_t>(len), aSize - 1);
```
Comment (superwhd) [inline src/lib/spinel/radio_spinel_impl.hpp:2591]: May `Clamp` fit this case better?

Comment (Irving-cl) [reply src/lib/spinel/radio_spinel_impl.hpp:2591]: I agree `Clamp` fits this case. But I think the comparison of `len < 0` here is not for a numeric meaning. It's to indicate that some error happened. So I think we'd better keep it here. Thoughts?

**Review by abtink — APPROVED** [review]
LGTM. Thanks 👍

**Review by superwhd — APPROVED** [review]
LGTM 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**